modules/swagger-codegen/src/main/java/io/swagger/codegen/languages/JavaCXFServerCodegen.java

```diff
@@ -36,14 +64,98 @@ public JavaCXFServerCodegen()
+            
+        	this.setUseSwaggerFeature(convertPropertyToBooleanAndWriteBack(USE_SWAGGER_FEATURE));
+            this.setUseWadlFeature(convertPropertyToBooleanAndWriteBack(USE_WADL_FEATURE));
+            this.setUseMultipartFeature(convertPropertyToBooleanAndWriteBack(USE_MULTIPART_FEATURE));
+            this.setUseGzipFeature(convertPropertyToBooleanAndWriteBack(USE_GZIP_FEATURE));
+            this.setUseLoggingFeature(convertPropertyToBooleanAndWriteBack(USE_LOGGING_FEATURE));
+            
+            boolean useBeanValidationFeature = convertPropertyToBooleanAndWriteBack(USE_BEANVALIDATION_FEATURE);
+            this.setUseBeanValidationFeature(useBeanValidationFeature);
+            if (useBeanValidationFeature) {
+            	LOGGER.info("make sure your target server supports Bean Validation 1.1");
+            }
+            
+            this.setGenerateSpringBootApplication(convertPropertyToBooleanAndWriteBack(GENERATE_SPRING_BOOT_APPLICATION));
+        }
+        
+       
+        supportingFiles.clear(); // Don't need extra files provided by AbstractJAX-RS & Java Codegen
+        
+        writeOptional(outputFolder, new SupportingFile("server/pom.mustache", "", "pom.xml"));
+        
+        if (this.generateSpringApplication) {
+        	writeOptional(outputFolder, new SupportingFile("server/readme.md", "", "readme.md"));
+            
+        	writeOptional(outputFolder, new SupportingFile("server/ApplicationContext.xml.mustache",
+                    ("src/main/resources"), "ApplicationContext.xml"));
+            writeOptional(outputFolder, new SupportingFile("server/web.mustache",
+                    ("src/main/webapp/WEB-INF"), "web.xml"));
+            writeOptional(outputFolder, new SupportingFile("server/context.xml.mustache",
+                    ("src/main/webapp/WEB-INF"), "context.xml"));
+            
+            // Jboss
+            writeOptional(outputFolder, new SupportingFile("server/jboss-web.xml.mustache",
+                    ("src/main/webapp/WEB-INF"), "jboss-web.xml"));
+            
+            // Spring Boot
+            if (this.generateSpringBootApplication) {
+            	writeOptional(outputFolder, new SupportingFile("server/SpringBootApplication.mustache",
+                        (testFolder + '/' + apiPackage).replace(".", "/"), "SpringBootApplication.java"));
+                	
+            }
+            
+        }
+        
+        
+    }
+
+
+	private boolean convertPropertyToBooleanAndWriteBack(String propertyKey) {
+		boolean booleanValue = false;
+		if (additionalProperties.containsKey(propertyKey)) {
+			booleanValue = Boolean.valueOf(additionalProperties.get(propertyKey).toString());
+		    // write back as boolean
+		    additionalProperties.put(propertyKey, booleanValue);
+		}
+		
+		return booleanValue;
+	} 
+    
+    @Override
+    public String getName()
+    {
+        return "jaxrs-cxf-server";
```

Comment:
@jfiala  Shall we keep `jaxrs-cxf` and for the client, rename it to jaxrs-cxf-client` instead? (to make this PR backward compatible)